src/Components/Blazor/Server/src/Builder/BlazorHostingEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,179 @@
+        /// Blazor application specified by <paramref name="clientAssemblyFilePath"/>.
+        /// </summary>
+        /// <param name="clientAssemblyFilePath">The file path of the client-side Blazor application assembly.</param>
+        /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/>.</param>
+        /// <returns>The <see cref="IApplicationBuilder"/>.</returns>
+        /// <remarks>
+        /// <para>
+        /// This method is intended to handle cases where URL path of the request does not contain a filename, and no other
+        /// endpoint has matched. This is convenient for routing requests for dynamic content to the client-side blazor
+        /// application, while also allowing requests for non-existent files to result in an HTTP 404.
+        /// </para>
+        /// </remarks>
+        public static IEndpointConventionBuilder MapFallbackToClientSideBlazor(this IEndpointRouteBuilder endpoints, string clientAssemblyFilePath, string filePath)
+        {
+            if (endpoints == null)
+            {
+                throw new ArgumentNullException(nameof(endpoints));
+            }
+
+            if (clientAssemblyFilePath == null)
+            {
+                throw new ArgumentNullException(nameof(clientAssemblyFilePath));
+            }
+
+            if (filePath == null)
+            {
+                throw new ArgumentNullException(nameof(filePath));
+            }
+
+            return MapFallbackToClientSideBlazor(endpoints, clientAssemblyFilePath, FallbackEndpointRouteBuilderExtensions.DefaultPattern, filePath);
+        }
+
+        /// <summary>
+        /// Adds a low-priority endpoint that will serve the the file specified by <paramref name="filePath"/> from the client-side
+        /// Blazor application specified by <typeparamref name="TClientApp"/>.
+        /// </summary>
+        /// <typeparam name="TClientApp">A type in the client-side application.</typeparam>
+        /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/>.</param>
+        /// <param name="pattern">The route pattern to match</param>
+        /// <param name="filePath">The relative path to the entry point of the client-side application.</param>
+        /// <returns>The <see cref="IApplicationBuilder"/>.</returns>
+        /// <remarks>
+        /// <para>
+        /// This method is intended to handle cases where URL path of the request does not contain a filename, and no other
+        /// endpoint has matched. This is convenient for routing requests for dynamic content to the client-side blazor
+        /// application, while also allowing requests for non-existent files to result in an HTTP 404.
+        /// </para>
+        /// </remarks>
+        public static IEndpointConventionBuilder MapFallbackToClientSideBlazor<TClientApp>(this IEndpointRouteBuilder endpoints, string pattern, string filePath)
```

Comment:
`pattern` isn't used anywhere in the method, besides the null check.

It would be possible to add E2E tests for these methods by having a bunch of different `Startup` classes in `BasicTestApp`, and then `TestServer` could map different subdirectories to different startup classes. Currently it maps `/subdir` to the main `Startup` class.

Reply:
fixed the bug. 

Will add tests for these all.